data-science-onramp/ai-platform/sklearn_test.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https: // www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import tarfile
+import time
+import uuid
+
+from google.cloud import storage
+from google.cloud.aiplatform import gapic as aip
+
+import pytest
+
+STAGING_BUCKET = f"sklearn-job-dir-{uuid.uuid4()}"
+INPUT_DIR = "inputs"
+TRAINER_DIR = "modules"
+MODEL_DIR = "model"
+TRAIN_DATA = "train.csv"
+TRAINER_TAR = "trainer.tar.gz"
+TRAIN_DATA_PATH = f"gs://{STAGING_BUCKET}/{INPUT_DIR}/{TRAIN_DATA}"
+
+PROJECT_ID = os.environ["GOOGLE_CLOUD_PROJECT"]
+REGION = "us-central1"
+MODEL_NAME = f"sklearn-test-{uuid.uuid4()}"
+JOB_ID = f"sklearn_{str(uuid.uuid4())[:7]}"
+DEPLOY_IMAGE = "gcr.io/cloud-aiplatform/training/scikit-learn-cpu.0-23:latest"
+
+TERMINAL_STATES = [
+    aip.JobState.JOB_STATE_SUCCEEDED,
+    aip.JobState.JOB_STATE_FAILED,
+    aip.JobState.JOB_STATE_CANCELLING,
+    aip.JobState.JOB_STATE_CANCELLED
+]
+
+
+@pytest.fixture
+def shared_state():
+    state = {}
+    yield state
+
+
+@pytest.fixture(autouse=True)
+def setup_teardown(shared_state):
+    storage_client = storage.Client()
+    bucket = storage_client.create_bucket(STAGING_BUCKET, location=REGION)
+    bucket.blob(f"{INPUT_DIR}/{TRAIN_DATA}").upload_from_filename(TRAIN_DATA, timeout=600)
+
+    with tarfile.open(TRAINER_TAR, mode="x:gz") as tar:
+        tar.add(f"{TRAINER_DIR}/")
+
+    bucket.blob(TRAINER_TAR).upload_from_filename(TRAINER_TAR)
+
+    aip_job_client = aip.JobServiceClient(
+        client_options={
+            "api_endpoint": f"{REGION}-aiplatform.googleapis.com"
+        }
+    )
+
+    yield bucket, aip_job_client
+
+    [blob.delete() for blob in bucket.list_blobs()]
+    bucket.delete()
```

Comment:
you either need to add the force true flag or wrap this in a try/except for a NotFound

Reply:
Ack.